spec/core/matchers/matchersUtilSpec.js

```diff
@@ -93,6 +93,24 @@ describe("matchersUtil", function() {
       expect(jasmineUnderTest.matchersUtil.equals(new Error("foo"), new Error("bar"))).toBe(false);
     });
 
+    it("fails for objects with different constructors", function() {
+      function One() {}
+      function Two() {}
+
+      expect(jasmineUnderTest.matchersUtil.equals(new One(), new Two())).toBe(false);
+    });
+
+    if (typeof document === 'object') {
+      it("passes for equivalent objects from different frames", function() {
```

Comment:
Also, I leaned towards not calling `it` if there's no `document` rather than just terminating the test early, because if you do the latter you get console warnings about specs that have no assertions.